meshkernel/py_structures.py

```diff
@@ -178,6 +179,21 @@ def __init__(
         self.geometry_separator: float = geometry_separator
         self.inner_outer_separator: float = inner_outer_separator
 
+        if len(self.x_coordinates) != len(self.y_coordinates):
+            raise InputError(
+                "The length of x_coordinates is not equal to the length of y_coordinates"
+            )
+
+        if len(self.values) > 0 and len(self.values) != len(self.x_coordinates):
+            raise InputError(
+                "The length of values is not equal to the length of x_coordinates"
+            )
+
+        if len(self.values) > 0 and len(self.values) != len(self.y_coordinates):
+            raise InputError(
+                "The length of  values is not equal to the length of y_coordinates"
+            )
+
```

Comment:
I don't think this test is necessary.
line 182 checks that x_coords and y_coords are the same length,  then
line 187 check values is same as x_coords (if length of values is > 0) 
at this point it must be that x_coords and y_coords are the same size.